### ISSUE TITLE
Not ignoring .git directory

### ISSUE THREAD

**Issue by igorvpcleao** [salesforce/p4-fusion#14]
Hi there,

After running a sync for a large project I found this error:
```
[ ERROR @ AddFileToIndex:147 ] GitAPI: -1:10: invalid path: 'redacted/redacted/redacted/redacted/redacted/redacted/redacted/redacted/Private/Interfaces/.git'
``` 
It seems that `p4-fusion` is trying to add `.git` from a submodule located in a deeper directory.

By looking at `p4-fusion` source code ([change_list.cc#L57](https://github.com/salesforce/p4-fusion/blob/master/p4-fusion/commands/change_list.cc#L57)), `p4-fusion` is ignoring `.git/*`, but not the `.git` per se. I believe this is a bug and it should also ignore files ending with `.git`.

Does it make sense? After implementing this change locally, the project was successfully synced.

**Comment by twarit-waikar** [comment]
@igorvpcleao Thanks for reporting this!

To double check, this `.git` is a file and not a directory in your case?

If yes, I think your reasoning is spot on. That should be the issue here. Feel free to open a PR with the changes that worked for you. It should be an easy fix (haven't thought about it 100%) so even if you are unable to do that, I think this is a valid issue that we can fix.

**Comment by igorvpcleao** [comment]
At first I thought it was a directory, but I've just double-checked and searched about submodules. `.git` here is indeed a file (used by submodules), not a directory.

Cool, sounds good. I'll proceed with the PR :).

Just a quick question. I'm not a c++ programmer and it looks like the language doesn't have an `ends_with` method for std::string on c++11 (only on c++20). As I have to look for `.git` at the end of the string, otherwise I may end up ignoring valid files with `.git` in the middle of it, should I create another file within `p4-fusion/utils/` to define this `ends_with` helper?

**Comment by twarit-waikar** [comment]
@igorvpcleao My initial idea was to use `std::string(path.end() - 4, path.end()) == ".git"` for this. It should probably be okay for now since we don't use such functionality a lot. Probably I could take it out in a future refactoring.

For submodules, the `.git` is created as a file instead of a folder like in a normal Git repository, which is basically a Git implementation detail at this point which p4-fusion is not handling properly.

**Comment by twarit-waikar** [comment]
Yup, double checked, this should work - https://godbolt.org/z/W6cdTd8K6

**Comment by twarit-waikar** [comment]
One trivial thing to check before performing this comparison would be to ensure the string is at least 4 characters long

**Comment by igorvpcleao** [comment]
@twarit-waikar thanks for checking it. Yeah, I did something similar here for testing :)
For the PR, I maintained your syntax, tho. Also, not sure how relevant is this, but I kept `/.git` in the check (instead of `.git`)

Please let me know if there's something you'd like me to change, or if the target branch is incorrect.

**Comment by igorvpcleao** [comment]
Ah, most important, please mind the equality was changed to `!=`, and the if here changes `fileData.shouldCommit = true`

**Comment by twarit-waikar** [comment]
Closed #14 via #15
